tensorboard/webapp/theme/_tb_theme.template.scss

```diff
@@ -68,12 +92,101 @@ $tb-dark-background: map-get($tb-dark-theme, background);
   }
 }
 
+@mixin _interal-tb-theme-prop-color(
+  $object-name,
+  $foreground-or-background,
+  $dark-foreground-or-background,
+  $property,
+  $name,
+  $value
+) {
+  $light-color: map-get($foreground-or-background, $name);
+  $dark-color: map-get($dark-foreground-or-background, $name);
+
+  @if (not $light-color) {
+    @error "Name '#{$name}' is not found in light #{$object-name}.";
+  }
+  @if (not $dark-color) {
+    @error "Name '#{$name}' is not found in dark #{$object-name}.";
+  }
+
+  #{$property}: $value $light-color;
+
+  @include tb-dark-theme {
+    #{$property}: $value $dark-color;
+  }
+}
+
+/**
+ * Styles an Angular component for a given CSS property using foreground theming
+ * dictionary. This mixin also facilitates theming for the dark mode.
+ *
+ * Example usage:
+ *
+ * div {
+ *   color: red;
+ *   @include tb-theme-foreground-prop(border-left, border, 5px solid);
+ *   // `tb-foreground` has the `border` key. Above generates:
+ *   // div[ngcomponentweirdid] {
+ *   //   border-left: 1px solid map_get($tb-foreground, border);
+ *   // }
+ *   // :host-context(body.dark-mode) div[ngcomponentweirdid] {
+ *   //   border-left: 1px solid map_get($tb-dark-foreground, border);
+ *   // }
+ *   //
+ *   // Third argument is optional. In the example below, `color` only takes
+ *   // color value so it is not needed.
+ *   @include tb-theme-foreground-prop(color, secondary-text);
+ * }
+ */
+@mixin tb-theme-foreground-prop($property, $name, $value: null) {
```

Comment:
The example mentioned is great.

Upon first glance, I think there is a little bit of confusion between these param names.  Wdyt about giving these params more specific names or doing a JSDoc-like comment, something along the lines of:

- $property -> $css-property
- $name -> $tb-variable  (must be a key in $tb-theme)
- $value -> $css-value-prefix

?

Reply:
Done.